pandas/core/indexes/datetimelike.py

```diff
@@ -430,8 +430,8 @@ def argmax(self, axis=None, skipna=True, *args, **kwargs):
     # --------------------------------------------------------------------
     # Rendering Methods
 
-    def _format_with_header(self, header, **kwargs):
-        return header + list(self._format_native_types(**kwargs))
+    def _format_with_header(self, header, na_rep='NaN', **kwargs):
```

Comment:
This line seems to be making tests fail. The default was `NaN` in class `Index` so I used `NaN` here too, but tests are expecting it to be `NaT`. I will change it to NaT and hope mypy doesn't complains about it.